APE_Bayesian_modeling.rst

```diff
@@ -85,29 +88,56 @@ Implementation
 --------------
 
 The implementation should proceed roughly as ordered below.  Specifically,
-first the priors, then the statistics subsections.  The next two subsections,
-fitters and evidence may be developed in either order.
+first the :ref:`priors<Priors>`, then the :ref:`statistics<Statistics>` subsections.
+The next two subsections, :ref:`fitters<Fitters>` and :ref:`evidence<Evidence>`
+may be developed in either order.
 
 Priors
 ======
 
 Three different ways of specifying a prior on a model should be supported.
 
-* The existing `bounds` attribute of `Parameter` should be interpreted as
-  a flat prior between the min/max values given in `bounds`.
-* The `prior` attribute of the `Parameter` class can be set to any 1D function
+* The existing ``bounds`` attribute of ``Parameter`` should be interpreted as
+  a flat prior between the min/max values given in ``bounds``.
+* The ``prior`` attribute of the ``Parameter`` class can be set to any 1D function
   that takes a single value and returns the prior evaluated at that value.
   This could be any astropy 1D model or a user written function.
 * For more complicated priors that simultaneously depend on multiple parameters,
-  a user written function that takes all the parameters as input and returns
-  the prior evaluated at those parameters.
+  can be defined as a user written function that takes all the parameters as input
+  and returns the prior evaluated at those parameters.
 
-The implementation of the 1st two ways will be done by having a generic prior
-function that is like the 3rd way that takes all the parameters as input.
-This generic prior function then uses any `Parameter.prior` functions and then
-any `Parameter.bounds` implied flat prior functions.  For any `Parameters`
+The implementation of the first two ways will be done by having a generic ``prior``
+function similar to the third way that takes all the parameters as input.
+This generic ``prior`` function then uses any ``Parameter.prior`` functions and then
+any ``Parameter.bounds`` implied flat prior functions.  For any ``Parameters``
 without any prior, this function would just give a constant prior (e.g., 1).
 
+
+class Parameter:
+    ...
+    self._prior = <user_supplied_prior>
+
+    @property
+    def prior(self):
+        if self._prior is not None:
+            return self._prior
```

Comment:
I had not been thinking of the user supplying a prior on each parameter, but now that I see this I like this option.